unit_tests/util/io.cpp

```diff
@@ -28,4 +30,77 @@ BOOST_AUTO_TEST_CASE(io_data)
     BOOST_CHECK_EQUAL_COLLECTIONS(data_out.begin(), data_out.end(), data_in.begin(), data_in.end());
 }
 
+BOOST_AUTO_TEST_CASE(io_nonexistent_file)
+{
+    static std::string IO_NONEXISTENT_FILE = "non_existent_test_io.tmp";
+
+    try {
+      osrm::storage::io::FileReader infile(IO_NONEXISTENT_FILE,
+                                           osrm::storage::io::FileReader::VerifyFingerprint);
+      BOOST_REQUIRE_MESSAGE( false, "Should not get here");
+    } catch (const osrm::util::exception &e) {
+      std::cout << e.what() << std::endl;
+      BOOST_REQUIRE(std::string(e.what()) == "Error opening non_existent_test_io.tmp:No such file or directory");
+    }
```

Comment:
I think there is a macro CHECK_THROWS of something like that for this exact use-case